taipy/core/_entity/_reload.py

```diff
@@ -41,11 +42,14 @@ def _reload(self, manager: str, obj):
         return entity
 
     def __enter__(self):
+        self._context_depth += 1
         self._no_reload_context = True
         return self
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        self._no_reload_context = False
+        self._context_depth -= 1
+        if self._context_depth == 0:
+            self._no_reload_context = False
```

Comment:
Do we need to have a lock when updating the `_context_depth`?

Reply:
Yes we do. I just forgot to do it.